### PR TITLE
PR for group without useless directory

### PR DESCRIPTION
**Pull request by Casmo888**

### CODE REVIEW

File: group/views.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        {'name':'44444', 'time':'20133333'},
+        {'name':'55555', 'time':'20144444'},
+        {'name':'66666', 'time':'20166666'},
+        {'name':'33333', 'time':'20122222'},
+        {'name':'44444', 'time':'20133333'},
+        {'name':'55555', 'time':'20144444'},
+        {'name':'66666', 'time':'20166666'},
+        {'name':'77777', 'time':'20177777'}
+    ]
+    if len(contest_data)>5 :
```
Comment (nickelcheng) [inline group/views.py]: You can query only 5 contest data instead of query all data and filter.

File: group/views.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        {'name':'ddddd', 'time':'20133333'},
+        {'name':'eeeee', 'time':'20144444'},
+        {'name':'fffff', 'time':'20166666'},
+        {'name':'ccccc', 'time':'20122222'},
+        {'name':'ddddd', 'time':'20133333'},
+        {'name':'eeeee', 'time':'20144444'},
+        {'name':'fffff', 'time':'20166666'},
+        {'name':'ggggg', 'time':'20177777'}
+    ]
+    if len(archive_data)>5 :
```
Comment (nickelcheng) [inline group/views.py]: You can query only 5 contest data instead of query all data and filter.

File: group/views.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    annowence_data = [
+        {'name':'annowence_1','op':'drowsy', 'time':'20111010'},
+        {'name':'annowence_2','op':'drowsy', 'time':'20140505'},
+        {'name':'annowence_3','op':'henry', 'time':'20122222'},
+        {'name':'annowence_4','op':'drowsy', 'time':'20133333'},
+        {'name':'annowence_5','op':'henry', 'time':'20144444'},
+        {'name':'annowence_6','op':'henry', 'time':'20166666'},
+        {'name':'annowence_7','op':'drowsy', 'time':'20177777'}
+    ]
+    if len(annowence_data)>3 :
```
Comment (nickelcheng) [inline group/views.py]: You can query only 5 contest data instead of query all data and filter.

File: group/views.py (hunk 1 above)
Comment (nickelcheng) [inline group/views.py]: Please fix this.

Comment (Casmo888) [inline group/views.py]: error Fixed in latest commit, thanks for reviewing

**Comment by Casmo888** [comment]
Problem fixed in views.py in the latest version, thanks for reviewing!

File: group/views.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.'''
+from django.http import HttpResponse
+from django.shortcuts import render_to_response, render
+def viewall_contest(request):
+        allData = [
```
Comment (nickelcheng) [inline group/views.py]: I prefer to rename allData to all_running_contest

File: group/views.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.'''
+from django.http import HttpResponse
+from django.shortcuts import render_to_response, render
+def viewall_contest(request):
```
Comment (nickelcheng) [inline group/views.py]: I prefer to rename this function to get_running_contest

File: group/views.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'contest', 'time':'20166666'},
+            {'name':'contest', 'time':'20177777'}
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': allData, 
+                'Title': 'contest',
+            })
+
+def viewall_archive(request):
```
Comment (nickelcheng) [inline group/views.py]: I prefer to rename this function to get_contest_archive

File: group/views.py [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'archive', 'time':'20144444'},
+            {'name':'archive', 'time':'20166666'},
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': allData, 
+                'Title': 'archive',
+            })
+
+def viewall_announce(request):
```
Comment (nickelcheng) [inline group/views.py]: I prefer to rename this function to get_all_announce.

File: group/views.py [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'contest', 'time':'20177777'}
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': allData, 
+                'Title': 'contest',
+            })
+
+def viewall_archive(request):
+        allData = [
```
Comment (nickelcheng) [inline group/views.py]: I prefer to rename allData to all_ended_contest

File: group/views.py [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'archive', 'time':'20166666'},
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': allData, 
+                'Title': 'archive',
+            })
+
+def viewall_announce(request):
+        allData = [
```
Comment (nickelcheng) [inline group/views.py]: I prefer to rename allData to all_announce.

**Comment by Casmo888** [comment]
fixed all naming rule followed the requirement.
Thanks for reviewing!

File: group/views.py [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+    archive_data = [
+        {'name':'aaaaa', 'time':'20111010'},
+        {'name':'bbbbb', 'time':'20140505'},
+        {'name':'ccccc', 'time':'20122222'},
+        {'name':'ddddd', 'time':'20133333'},
+        {'name':'eeeee', 'time':'20144444'},
+    ]
+
+    annowence_data = [
```
Comment (nickelcheng) [inline group/views.py]: typo?

File: group/views.py [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+    ]
+
+    ta_data =[
+        {'name': 'Tom'},
+        {'name': 'Amy'},
+    ]
+
+    return render(
+        request, 'group/groupDetail.html', {
+            'c_data': contest_data,
```
Comment (nickelcheng) [inline group/views.py]: I suggest use 'contest': contest_data for code readability.

File: group/views.py [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+    ta_data =[
+        {'name': 'Tom'},
+        {'name': 'Amy'},
+    ]
+
+    return render(
+        request, 'group/groupDetail.html', {
+            'c_data': contest_data, 
+            'a_data': archive_data,
```
Comment (nickelcheng) [inline group/views.py]: I suggest use 'ended_contest'.

File: group/views.py [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+            'an_data': annowence_data,
```
Comment (nickelcheng) [inline group/views.py]: I suggest use 'announce'.

File: group/views.py [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+            'ta_name': ta_data,
+            's_data': student_data,
```
Comment (nickelcheng) [inline group/views.py]: I suggest use 'students'.

File: manage.py [hunk 15]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
```
Comment (nickelcheng) [inline manage.py]: Please do not remove this.

File: group/views.py [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'announce', 'time':'20140505'},
+            {'name':'announce', 'time':'20122222'},
+            {'name':'announce', 'time':'20133333'},
+            {'name':'announce', 'time':'20144444'},
+            {'name':'announce', 'time':'20166666'},
+            {'name':'announce', 'time':'20122222'},
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': allData,
```
Comment (nickelcheng) [inline group/views.py]: I suggest use 'announce_list': contest_data for code readability.

File: group/views.py [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'archive', 'time':'20111010'},
+            {'name':'archive', 'time':'20140505'},
+            {'name':'archive', 'time':'20122222'},
+            {'name':'archive', 'time':'20133333'},
+            {'name':'archive', 'time':'20144444'},
+            {'name':'archive', 'time':'20166666'},
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': allData,
```
Comment (nickelcheng) [inline group/views.py]: I suggest use 'all_ended_contest_list': contest_data for code readability.

File: group/views.py [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'contest', 'time':'20166666'},
+            {'name':'contest', 'time':'20122222'},
+            {'name':'contest', 'time':'20133333'},
+            {'name':'contest', 'time':'20144444'},
+            {'name':'contest', 'time':'20166666'},
+            {'name':'contest', 'time':'20177777'}
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': allData,
```
Comment (nickelcheng) [inline group/views.py]: I suggest use 'all_running_contest_list'.

File: group/views.py [hunk 19]
```diff
@@ -0,0 +1,155 @@
...
+        {'name': 'GGgroup', 'ta_name': 'drowsy, henry', 'mem_number': '42', 'gid': '001'},
+        {'name': '2ndgroup', 'ta_name': 'TA1', 'mem_number': '1', 'gid': '002'},
+        {'name': '3rdgroup', 'ta_name': 'TA2, henry', 'mem_number': '22', 'gid': '003'},
+        {'name': '4thgroup', 'ta_name': 'drowsy, TA3', 'mem_number': '34', 'gid': '004'},
+        {'name': '5thgroup', 'ta_name': 'TA4', 'mem_number': '105', 'gid': '005'},
+        {'name': '6thgroup', 'ta_name': 'T5A', 'mem_number': '42', 'gid': '006'}
+    ]
+    return render(
+        request,'group/groupList.html', {
+            'G_data': group_data
```
Comment (nickelcheng) [inline group/views.py]: I suggest using 'group_list'.

File: group/views.py [hunk 20]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'announce', 'time':'20140505'},
+            {'name':'announce', 'time':'20122222'},
+            {'name':'announce', 'time':'20133333'},
+            {'name':'announce', 'time':'20144444'},
+            {'name':'announce', 'time':'20166666'},
+            {'name':'announce', 'time':'20122222'},
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': all_announce,
```
Comment (nickelcheng) [inline group/views.py]: I suggest using 'all_contest_list'.

File: group/views.py [hunk 21]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'archive', 'time':'20111010'},
+            {'name':'archive', 'time':'20140505'},
+            {'name':'archive', 'time':'20122222'},
+            {'name':'archive', 'time':'20133333'},
+            {'name':'archive', 'time':'20144444'},
+            {'name':'archive', 'time':'20166666'},
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': all_ended_contest,
```
Comment (nickelcheng) [inline group/views.py]: I suggest using 'all_contest_list'.

File: group/views.py [hunk 22]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'contest', 'time':'20166666'},
+            {'name':'contest', 'time':'20122222'},
+            {'name':'contest', 'time':'20133333'},
+            {'name':'contest', 'time':'20144444'},
+            {'name':'contest', 'time':'20166666'},
+            {'name':'contest', 'time':'20177777'}
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': all_running_contest,
```
Comment (nickelcheng) [inline group/views.py]: I suggest using 'all_contest_list'.

File: group/views.py [hunk 23]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'contest', 'time':'20122222'},
+            {'name':'contest', 'time':'20133333'},
+            {'name':'contest', 'time':'20144444'},
+            {'name':'contest', 'time':'20166666'},
+            {'name':'contest', 'time':'20177777'}
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': all_running_contest, 
+                'Title': 'contest',
```
Comment (nickelcheng) [inline group/views.py]: I prefer 'title' to 'Title'.

File: group/views.py [hunk 24]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'archive', 'time':'20140505'},
+            {'name':'archive', 'time':'20122222'},
+            {'name':'archive', 'time':'20133333'},
+            {'name':'archive', 'time':'20144444'},
+            {'name':'archive', 'time':'20166666'},
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': all_ended_contest, 
+                'Title': 'archive',
```
Comment (nickelcheng) [inline group/views.py]: I prefer 'title' to 'Title'.

File: group/views.py [hunk 25]
```diff
@@ -0,0 +1,155 @@
...
+            {'name':'announce', 'time':'20122222'},
+            {'name':'announce', 'time':'20133333'},
+            {'name':'announce', 'time':'20144444'},
+            {'name':'announce', 'time':'20166666'},
+            {'name':'announce', 'time':'20122222'},
+        ] 
+        return render(
+            request, 'group/viewall.html', {
+                'C_data': all_announce, 
+                'Title': 'announce',
```
Comment (nickelcheng) [inline group/views.py]: I prefer 'title' to 'Title'.

File: group/templates/group/groupList.html [hunk 26]
```diff
@@ -0,0 +1,141 @@
...
+<!--1. Modal for search group-->
+<div class="modal fade" id="searchGroupModal" tabindex="-1" role="dialog" 
+     aria-labelledby="myModalLabel" aria-hidden="true">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="btn-default" class="close" data-dismiss="modal" aria-hidden="true">
+          &times; 
+        </button>
+        <h4 class="modal-title" id="myModalLabel">Search</h4>
```
Comment (bruce3557) [inline group/templates/group/groupList.html]: Any more meaningful id of this text?
I prefer each id of html has its own meaning.

**Comment by bruce3557** [comment]
Besides naming issue, I have no suggestion.
Thanks for your hard work!

**Comment by Casmo888** [comment]
This commit for naming rule(Remove useless id, change parameter's name to more meaningful one)
Also change contest time to (start time + end time)
Thanks for reviewing and all comments!

**Comment by nickelcheng** [comment]
This PR is ok for me. Thanks for @Casmo888 's effort.
Does @jim800121 have other suggestions?

**Comment by nickelcheng** [comment]
Please merge latest code then this PR can be merged.

**Comment by jim800121** [comment]
It's good for me.

**Comment by Casmo888** [comment]
Merged with latest code.
This pull request can now be automatically merged by project collaborators.

**Comment by nickelcheng** [comment]
Going to merge.

**Pull request merged**